machine.go

```diff
@@ -1171,7 +1171,8 @@ func (m *Machine) CreateSnapshot(ctx context.Context, memFilePath, snapshotPath
 // loadSnapshot loads a snapshot of the VM
 func (m *Machine) loadSnapshot(ctx context.Context, snapshot *SnapshotConfig) error {
 	snapshotParams := &models.SnapshotLoadParams{
-		MemFilePath:         &snapshot.MemFilePath,
+		MemFilePath:         snapshot.MemFilePath,
```

Comment:
I was confused why making `mem_file_path` optional caused `MemFilePath` to now be a `string` instead of `*string`. I would have guessed it'd be the other way around - optional should imply `*string` and required should imply `string`, no?

I don't think this is a big deal, was just curious if anyone has any idea why this happens.